garrysmod/lua/includes/modules/outline.lua

```diff
@@ -0,0 +1,120 @@
+module("outline",package.seeall)
+
+local List,ListSize = {},0
+local RenderEnt = NULL
+
+local CopyMat		= Material("pp/copy")
+local OutlineMat	= CreateMaterial("OutlineMat","UnlitGeneric",{["$ignorez"] = 1,["$alphatest"] = 1})
+local StoreTexture	= render.GetScreenEffectTexture(0)
+local DrawTexture	= render.GetScreenEffectTexture(1)
+
+local ENTS,COLOR,MODE = 1,2,3
+local MODE_NOTVISIBLE,MODE_VISIBLE = 1,2
+
+function Add(ents,color,mode)
+	if ListSize>=255 then return end		--Maximum 255 reference values
+	if !istable(ents) then ents = {ents} end	--Support for passing Entity as first argument
+	if ents[1]==nil then return end			--Do not pass empty tables
+	
+	table.insert(List,{
+		[ENTS] = ents,
+		[COLOR] = color,
+		[MODE] = mode
+	})
+	
+	ListSize = ListSize+1
+end
+
+function RenderedEntity()
+	return RenderEnt
+end
+
+local function Render()
+	local scene = render.GetRenderTarget()
+	render.CopyRenderTargetToTexture(StoreTexture)
+	
+	local w,h = ScrW(),ScrH()
+	
+	render.Clear(0,0,0,0,true,true)
+
+	render.SetStencilEnable(true)
+		cam.IgnoreZ(true)
+		render.SuppressEngineLighting(true)
+	
+		render.SetStencilWriteMask(255)
+		render.SetStencilTestMask(255)
+		
+		render.SetStencilCompareFunction(STENCIL_ALWAYS)
+		render.SetStencilFailOperation(STENCIL_KEEP)
+		render.SetStencilZFailOperation(STENCIL_REPLACE)
+		render.SetStencilPassOperation(STENCIL_REPLACE)
+		
+		cam.Start3D()
+			for k,v in ipairs(List) do
```

Comment:
Now that you have a ListSize variable, you can make this loop
```
for i = 1, ListSize do
   local v = List[i]
   render.SetStencilReferenceValue(i)
   ...
```

for slightly more efficient iteration. Same with the cam 2D loop.